### PR TITLE
Use Constants for Class Status Instead of Hard-coded Numbers

### LINKED ISSUE #2996: Use constants for class status instead of hard-coded values
We have hard-coded class status numbers 10, 0, 5, -10, -20 throughout the codebase. This is very hard to read. We should use the constants defined at the top of `esp/esp/programs/models/class_.py` instead.

### PR DESCRIPTION
**Pull request by AlpacaFur**
This PR creates an enum-like class with the constant numbers we use for class status (cancelled, rejected, unreviewed, hidden, accepted) to make the code more readable.

Fixes #2996.

### CODE REVIEW

File: esp/esp/program/modules/handlers/jsondatamodule.py [hunk 1]
```diff
@@ -769,11 +770,11 @@ def class_nums(prog):
         class_num_list.append(("Total # of Classes Scheduled", classes.filter(sections__meeting_times__isnull=False).distinct().count()))
         class_num_list.append(("Total # of Class Sections", sections.distinct().count()))
         class_num_list.append(("Total # of Class Sections Scheduled", sections.filter(meeting_times__isnull=False).distinct().count()))
-        class_num_list.append(("Total # of Lunch Classes", classes.filter(category__category = "Lunch").filter(status=10).distinct().count()))
-        class_num_list.append(("Total # of Classes <span style='color: #00C;'>Unreviewed</span>", classes.filter(status=0).exclude(category__category='Lunch').distinct().count()))
-        class_num_list.append(("Total # of Classes <span style='color: #0C0;'>Accepted</span>", classes.filter(status=10, sections__status=10).exclude(category__category='Lunch').distinct().count()))
```
Comment (hwatheod) [inline esp/esp/program/modules/handlers/jsondatamodule.py:774]: You missed one here, `sections__status=10`

Comment (hwatheod) [reply esp/esp/program/modules/handlers/jsondatamodule.py:774]: Line 774

Comment (AlpacaFur) [reply esp/esp/program/modules/handlers/jsondatamodule.py:774]: Good catch, it's now fixed 👍

**Review by hwatheod — APPROVED** [review]
Looks good! Thank you for doing this!

**Pull request merged**